src/SA/locale/entry.c

```diff
@@ -0,0 +1,36 @@
+#include <windows.h>
+#include "bofdefs.h"
+#include "base.c"
+#include "anticrash.c"
+
+DECLSPEC_IMPORT int WINAPI KERNEL32$GetLocaleInfoEx(LPCWSTR lpLocaleName, LCTYPE LCType, LPWSTR lpLCData, int cchData);
+WINBASEAPI int WINAPI KERNEL32$GetSystemDefaultLocaleName(LPCWSTR lpLocaleName, int cchLocaleName);
+DECLSPEC_IMPORT LCID WINAPI KERNEL32$LocaleNameToLCID(LPCWSTR lpName, DWORD dwFlags);
+DECLSPEC_IMPORT int WINAPI KERNEL32$GetDateFormatEx(LPCWSTR lpLocaleName, DWORD dwFlags, const SYSTEMTIME *lpData, LPCWSTR lpFormat, LPWSTR lpDateStr, int cchDate, LPCWSTR lpCalendar);
+VOID go( 
+	IN PCHAR Buffer, 
+	IN ULONG Length 
+) 
+{
+
+	int iResult;
+	int BUFFER_SIZE = 512;
+	WCHAR name[BUFFER_SIZE];
+	WCHAR wcBuffer[BUFFER_SIZE];
+	WCHAR sysTime[BUFFER_SIZE];
+	WCHAR geoid[80];
+	iResult = KERNEL32$GetSystemDefaultLocaleName(name, BUFFER_SIZE);
+	if(iResult > 0) {
+		iResult = KERNEL32$GetLocaleInfoEx(name, LOCALE_SENGLANGUAGE, wcBuffer, BUFFER_SIZE);
```

Comment:
While I agree if the first one worked these very likely will as well, please check the return values against failure before passing the value to the next call.

Reply:
will do.